Add blueprint v2 docs (#77)
cloudogu/k8s-blueprint-operator: docs/operations/blueprintV2_format_en.md

# Blueprint format

The blueprint offers the option of adding or removing Dogus, Components and configurations.
Special configurations can also be defined for Dogus and components.
These are written to the corresponding Dogu and component CRs and are not saved in the EcoSystem registry.

All fields of the blueprint are described below and illustrated with examples.

## BlueprintApi

> * Not optional

Reviewer: `Required` would be more concise

* Data type: string
* Content: The `blueprintApi` field specifies the API version of the blueprint.
* Example: `"blueprintApi": "v2"`

## Dogus

* Not optional
* Data type: Array<Dogu>
* Content: The `dogus` field is a list of Dogus and describes the status of the Dogus in the system.
* Example:
```
"dogus": [
  {
    "name": "official/postgresql",
    "targetState": "present",
    "version": "12.15-2"
  }
]
```

### Dogu

A Dogu can contain the following fields:

#### Name

* Not optional
* Data type: string

> * Content: Specifies the name including namespace of the Dogu.

Reviewer: ```suggestion
* Content: Specifies the name including the namespace of the Dogu.
```

* Example: `"name": "official/cas"`

#### TargetState

* Not optional
* Data type: string
* Content: Specifies whether a Dogu should be present or not.
* Example: `"targetState": "present"` or `"targetState": "absent"`

#### Version

* Optional for `targetState=absent`. Not optional for `targetState=present`.
* Data type: string
* Content: Specifies the version of the Dogu.
* Example: `"version": "12.15-2"`

#### PlatformConfig

The `platformConfig` field offers the option of transferring specific configurations for the execution platform (e.g. Kubernetes).
This configuration can be used to define resources and reverse proxy configurations.

##### Resource.minVolumeSize

* Optional
* Data type: string

> * Content: Specifies the minimum volume size of a Dogu. If the current volume is smaller, a volume increase is performed. The unit must be specified with the binary prefix (e.g. `Mi` or `Gi`).

Reviewer: Maybe add a remark that shrinking volumes is not supported.

> * Content: Specifies the minimum volume size of a Dogu. If the current volume is smaller, a volume increase is performed. The unit must be specified with the binary prefix (e.g. `Mi` or `Gi`).

Reviewer: Not a prefix.
```suggestion
* Content: Specifies the minimum volume size of a Dogu. If the current volume is smaller, a volume increase is performed. The unit must be specified with the binary suffix (e.g. `Mi` or `Gi`).
```

* Example:
```
"dogus": [
  {
    "name": "official/nexus",
    "targetState": "present",
    "version":"3.59.0-2",
    "platformConfig": {
      "resource": {
        "minVolumeSize": "5Gi"
      }
    }
  }
]
```

> The Dogu-Operator creates Dogus with 2Gi volumes. The Nexus Dogu requires a larger volume and must be configured via this entry.

##### ReverseProxy.maxBodySize

* Optional
* Data type: string

> * Content: Specifies the maximum file size for HTTP request (`0`=unlimited). The unit must be specified with the decimal prefix (e.g. `M` or `G`).

Reviewer: Not a prefix.
```suggestion
* Content: Specifies the maximum file size for HTTP request (`0`=unlimited). The unit must be specified with the decimal suffix (e.g. `M` or `G`).
```

]
```

##### ReverseProxy.rewriteTarget

* Optional
* Data type: string
* Content: Defines a rewrite target.
* Example:
```
"dogus": [
  {
    "name": "official/postgresql",
    "targetState": "present",
    "version": "12.15-2",
    "platformConfig": {
      "reverseProxy": {
        "rewriteTarget": "/"
      }
    }
  }
]
```

##### ReverseProxy.additionalConfig

* Optional
* Data type: string
* Content: Adds any additional proxy configuration.
* Example:
```
"dogus": [
  {
    "name": "official/postgresql",
    "targetState": "present",
    "version": "12.15-2",
    "platformConfig": {
      "reverseProxy": {
        "additionalConfig": "<config>"
      }
    }
  }
]
```

## Components

* Not optional
* Data type: Array
* Contents: The `components` field is a list of components and describes the status of the components in the system.
* Example:
```
"components": [
  {
    "name": "k8s/k8s-dogu-operator",
    "targetState": "present",
    "version": "1.0.1"
  },
  {
    "name": "k8s/k8s-dogu-operator-crd",
    "targetState": "present",
    "version": "1.0.1"
  }
}
```

### Component

A component can contain the following fields:

#### Name

* Not optional
* Data type: string

> * Content: Specifies the name including namespace of the component.

Reviewer: ```suggestion
* Content: Specifies the name including the namespace of the component.
```

* Example: `"name": "k8s/k8s-dogu-operator"`

#### TargetState

* Not optional
* Data type: string
* Content: Specifies whether a component should be present or not.
* Example: `"targetState": "present"` or `"targetState": "absent"`

#### Version

* Optional for `targetState=absent`. Not optional for `targetState=present`.
* Data type: string
* Content: Specifies the version of the component.
* Example: `"version": "12.15-2"`

#### DeployConfig

The `deployConfig` field offers the option of transferring specific configurations for the deployment of a component.

> This configuration can be used, for example, to define the component CR or any helm values.

Reviewer: ```suggestion
This configuration can be used to define certain fields of the component CR and among others it is therefore possible to set helm values.
```
Maybe like this?

Author: Yes this sounds better :+1:
